src/ts/components/subtitleoverlay.ts

```diff
@@ -4,30 +4,114 @@ import SubtitleCueEvent = bitmovin.PlayerAPI.SubtitleCueEvent;
  */
-export class SubtitleOverlay extends Container<ContainerConfig> {
+export class SubtitleOverlay extends Container<SubtitleOverlayConfig> {
+
+  public config: SubtitleOverlayConfig;
+
+  private subtitleManager: ActiveSubtitleManager;
+  private forceSubtitle: boolean = false;
+  private forcedSubtitle: SubtitleLabel = new SubtitleLabel({text: 'example subtitle'});
 
   private static readonly CLASS_CONTROLBAR_VISIBLE = 'controlbar-visible';
 
-  constructor(config: ContainerConfig = {}) {
+  constructor(config: SubtitleOverlayConfig = {}) {
     super(config);
 
-    this.config = this.mergeConfig(config, {
-      cssClass: 'ui-subtitle-overlay',
-    }, this.config);
+    this.config = this.mergeConfig( config, <SubtitleOverlayConfig>{
+        cssClass: 'ui-subtitle-overlay',
+        fontColor: ColorUtils.foreground,
+        backgroundColor: ColorUtils.background,
+        windowColor: ColorUtils.background,
+        fontFamily: '',
+        fontStyle: '',
+        fontVariant: 'normal',
+        characterEdge: '',
+        fontCoefficient: 1,
+        fontSize: 1.2,
+      }, this.config);
+
+    config = this.config;
+    if (StorageUtils.hasLocalStorage()) {
+      let store = window.localStorage;
+      let forcedSubtitleDom = this.forcedSubtitle.getDomElement();
+
+      let fontColor = store.getItem('fontColor');
+      if (fontColor != null) {
+        this.config.fontColor = ColorUtils.colorFromCss(fontColor, ColorUtils.foreground);
+      }
+      let backgroundColor = store.getItem('backgroundColor');
+      if (backgroundColor != null) {
+        this.config.backgroundColor = ColorUtils.colorFromCss(backgroundColor, ColorUtils.background);
+      }
+      let windowColor = store.getItem('windowColor');
+      if (windowColor != null) {
+        this.config.windowColor = ColorUtils.colorFromCss(windowColor, ColorUtils.background);
+      }
+      let fontFamily = store.getItem('fontFamily');
+      if (fontFamily != null) {
+        this.config.fontFamily = fontFamily;
+      }
+      let fontStyle = store.getItem('fontStyle');
+      if (fontStyle != null) {
+        this.config.fontStyle = fontStyle;
+      }
+      let fontVariant = store.getItem('fontVariant');
+      if (fontVariant != null) {
+        this.config.fontVariant = fontVariant;
+      }
+      let characterEdge = store.getItem('characterEdge');
+      if (characterEdge != null) {
+        this.config.characterEdge = characterEdge;
+      }
+      let fontCoefficient = store.getItem('fontCoefficient');
+      if (fontCoefficient != null) {
+        this.config.fontCoefficient = Number(fontCoefficient);
+      }
+    }
+    // This css property isn't applied to the subtitle cue
+    // and therefore shoud be applied now
+    this.getDomElement().css('background', this.config.windowColor.toCSS());
+    this.updateSubtitleLabelCss();
   }
 
   configure(player: bitmovin.PlayerAPI, uimanager: UIInstanceManager): void {
```

Comment:
In here you can get the global UI config and extract the `UISubtitleConfig` by calling `uimanager.getConfig().subtitle`